File: saved_files/sort.cpp
#include <iostream>
using namespace std;

void bauble(){
    int arr[] = {9, 8, 7, 6, 5, 4, 3, 2, 1, 0};

    int iter = 0;
    bool end = true;
    
    
    while(end){

        cout << iter++  << "\t:\t[ ";
        for(int i = 0; i < sizeof(arr)/sizeof(int); i++){
            cout << arr[i] << " ";
        }
        cout << "]\n";

        for(int i = 1; i < sizeof(arr)/sizeof(int); i++){
            end = false;
            if(arr[i-1] > arr[i]){
                int x = arr[i-1];
                arr[i-1] = arr[i];
                arr[i] = x;
                end = true;
                break;
            }
        }
    

    }


    cout << "[ ";
    for(int i = 0; i < sizeof(arr)/sizeof(int); i++){
        cout << arr[i] << " ";
    }
    cout << "]\n";


}


void sort2(){
    int arr[] = {9, 8, 7, 6, 5, 4, 3, 2, 1, 0};
    int iter = 0;
    int end = false;

    while(!end){
        for(int i = 0; i <  sizeof(arr)/sizeof(int); i++){
            int min;
            for(int i = iter+1; i <  sizeof(arr)/sizeof(int); i++){
                min = arr[i-1];
                if(arr[i] < min){
                    min = arr[i];
                }
            }

        }
    }


    cout << "[ ";
    for(int i = 0; i < sizeof(arr)/sizeof(int); i++){
        cout << arr[i] << " ";
    }
    cout << "]\n";

}


int main(){
    bauble();
    // sort2();
}   

File: saved_files/sort.py
arr = [9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
iter = 0

arrLen = len(arr)
for i in range(arrLen):
    min = arr[i]
    minIndex = 0
    for j in range(i, arrLen):
        if(min > arr[j]):
            min = arr[j]
            minIndex = j

    if(arr[i] != min):
        current = arr[i]
        arr[i] = arr[minIndex]
        arr[minIndex] = current
        print(str(iter) + "\t:\t",  end='')
        print(arr)
        iter += 1

print(arr)


# def shellSort(array):
# 	increment = len(array) // 2
# 	while increment > 0:

# 		for startPosition in range(increment):
# 			gapInsertionSort(array, startPosition, increment)

# 		print("После инкрементации размера на", increment,"массив:", array)

# 		increment //= 2

# def gapInsertionSort(array, low, gap):
    
# 	for i in range(low + gap, len(array), gap):
# 		currentvalue = array[i]
# 		position = i

# 		while position >= gap and array[position - gap] > currentvalue:
# 			array[position] = array[position - gap]
# 			position = position - gap

# 		array[position] = currentvalue


# shellSort([34,5, 56, 5, 67, 56, 567, 567, 87, 34, 23, 34,65, 7667, 78, 34, 33, 67, 89])
